### PR TITLE
build(*): optimize build process

### PR DESCRIPTION
**Pull request by DiyanDimitrov**
Changes:
- removed `src` folder from the publish build
- changed `lit` to hard dependency
- added `sideEffects: false` in the package.json
- removed themes scss from compiling to ts/js
- removed non-material scss files from compiling to ts/js

### CODE REVIEW

File: scripts/build-styles.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ async function sassRender(sourceFile, templateFile, outputFile) {
 
 (async () => {
   const template = path.resolve(process.argv[1], '../styles.tmpl');
-  const paths = await globby(['src/components/**/*.scss', 'src/styles/themes/**/*.scss']);
+  const paths = await globby(['src/components/**/*.material.scss']);
```
Comment (gedinakova) [inline scripts/build-styles.js:41]: Is this applicable to material only?

File: scripts/build.js [hunk 2]
```diff
@@ -32,7 +31,9 @@ async function buildThemes() {
       outCss = outCss.substr(1);
     }
 
-    await writeFile(DEST_DIR(sassFile.replace(/\.scss$/, '.css')), outCss, 'utf-8');
+    const outputFile = DEST_DIR(sassFile.replace(/\.scss$/, '.css').replace('src/', ''));
```
Comment (gedinakova) [inline scripts/build.js:34]: How about replacing "src/styles" so we end up with a path like product/themes/theme.file?

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: scripts/build-styles.js (hunk 1 above)
Comment (DiyanDimitrov) [reply scripts/build-styles.js:41]: Yes, the other component theme files are only used in the global theme files.

File: scripts/build.js (hunk 2 above)
Comment (DiyanDimitrov) [reply scripts/build.js]: Done.

**Review by gedinakova — APPROVED** [review]

**Review by gedinakova — APPROVED** [review]

**Review by rkaraivanov — APPROVED** [review]

**Pull request merged**